exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,623 @@
+          .addContext("User name", userName)
+          .build(logger);
+    }
+    reader = new BufferedReader(new InputStreamReader(in, Charsets.UTF_8));
+  }
+
+  private void projectAll() {
+    List<String> fields = new ArrayList<>();
+    fields.addAll(formatConfig.getFields());
+    for (int i = fields.size(); i < capturingGroups; i++) {
+      fields.add("field_" + i);
+    }
+    columns = new ColumnDefn[capturingGroups];
+
+    for (int i = 0; i < capturingGroups; i++) {
+      columns[i] = makeColumn(fields.get(i), i);
+    }
+
+
+  }
+
+  private void projectSubset() {
+    Collection<SchemaPath> project = this.getColumns();
+    assert !project.isEmpty();
+    columns = new ColumnDefn[project.size()];
+
+    List<String> fields = formatConfig.getFields();
+    int colIndex = 0;
+
+
+
+    for (SchemaPath column : project) {
+      if (column.getAsNamePart().hasChild()) {
+        throw UserException
+            .validationError()
+            .message("The log format plugin supports only simple columns")
+            .addContext("Projected column", column.toString())
+            .build(logger);
+      }
+      String name = column.getAsNamePart().getName();
+
+      //Need this to retrieve unnamed fields
+      Pattern r = Pattern.compile("^field_(\\d+)$");
+
+      // Now create matcher object.
+      Matcher m = r.matcher(name);
+      int patternIndex = -3;
+
+      if (m.find()) {
+        //if no fields are defined in the configuration, then all the fields have names of 'field_n'
+        //Therfore n is the column index
+        patternIndex = Integer.parseInt(m.group(1));
+      } else {
+        for (int i = 0; i < fields.size(); i++) {
+          if (fields.get(i).equalsIgnoreCase(name)) {
+            patternIndex = i;
+            break;
+          } else if( name.equals("_raw")){
+            patternIndex = -1;
+            break;
+          } else if( name.equals("_unmatched_rows")){
+            patternIndex = -2;
+            unmatchedColumnIndex = colIndex;
+          }
+        }
+      }
+      columns[colIndex++] = makeColumn(name, patternIndex);
+    }
+  }
+
+  private ColumnDefn makeColumn(String name, int patternIndex) {
+    String typeName = null;
+    if (patternIndex <= -1) {
+      // Use VARCHAR for missing columns
+      // (instead of Drill standard of nullable int)
+      typeName = MinorType.VARCHAR.name();
+    } else if (patternIndex < formatConfig.dataTypes.size()) {
+      typeName = formatConfig.dataTypes.get(patternIndex);
+    }
+    if (typeName == null) {
+      // No type name. VARCHAR is a safe guess
+      typeName = MinorType.VARCHAR.name();
+    }
+    if( name.equals("_raw")){
+      return new VarCharDefn(name, patternIndex);
+    }
+
+    MinorType type = MinorType.valueOf(typeName);
+    switch (type) {
+      case VARCHAR:
+        return new VarCharDefn(name, patternIndex);
+      case INT:
+      case SMALLINT:
+      case BIGINT:
```

Comment:
Isn't this a bit of a lie? The user expresses a data type, then we second-guess it here. If I ask for `BigInt`, I will be somewhat surprised that it only fits `Int` values. Similarly, below, if I ask for a `Decimal` value, I'll be surprised that I get a float, which can lose precision.

Should we be more strict about the types supported?